pkg/test/provisioner.go

```diff
@@ -59,25 +61,35 @@ func Provisioner(overrides ...ProvisionerOptions) *v1alpha5.Provisioner {
 	if options.Limits == nil {
 		options.Limits = v1.ResourceList{v1.ResourceCPU: resource.MustParse("1000")}
 	}
-	if options.Provider == nil {
-		options.Provider = struct{}{}
-	}
-	provider, _ := json.Marshal(options.Provider)
+
 	provisioner := &v1alpha5.Provisioner{
 		ObjectMeta: ObjectMeta(options.ObjectMeta),
 		Spec: v1alpha5.ProvisionerSpec{
 			Requirements:         options.Requirements,
 			KubeletConfiguration: options.Kubelet,
-			Provider:             &runtime.RawExtension{Raw: provider},
 			ProviderRef:          options.ProviderRef,
 			Taints:               options.Taints,
 			StartupTaints:        options.StartupTaints,
 			Labels:               options.Labels,
 			Limits:               &v1alpha5.Limits{Resources: options.Limits},
+			TTLSecondsAfterEmpty: ptr.Int64(10),
 		},
 		Status: options.Status,
 	}
+
+	if options.ProviderRef == nil {
+		if options.Provider == nil {
+			options.Provider = struct{}{}
+		}
+		provider, err := json.Marshal(options.Provider)
+		if err != nil {
+			panic(err)
+		}
+		provisioner.Spec.Provider = &runtime.RawExtension{Raw: provider}
+	}
 	provisioner.SetDefaults(context.Background())
-	_ = provisioner.Validate(context.Background())
+	if err := provisioner.Validate(context.Background()); err != nil {
+		logging.FromContext(context.TODO()).Info("TODO: Fix the tests that cause this")
```

Comment:
A ton of existing tests break if I fail here, since the error was originally squashed incorrectly. I wanted to log and clean it up later, since I'm OOTO for a few days and wanted to unblock folks.